### ISSUE TITLE
Select2 Hidden in Update-Modal

### ISSUE THREAD

**Issue by Squarius** [kartik-v/yii2-widget-select2#70]
Its set to display: none; if u use it with ajaxcrud in modal:

![select2](https://cloud.githubusercontent.com/assets/5981582/9971749/8af6b7c8-5e5e-11e5-90c5-efd031af3853.jpg)

How may i change this default behavior?

**Comment by siliconalchemy** [comment]
Same for me too, calling select2 in activeform loaded by ajax into a modal.

**Comment by siliconalchemy** [comment]
Actually it seems like including a kartik-select2 form widget also breaks the yii form js validation, although no js errors in the console.
All works fine within a normal non-modal form.

**Comment by siliconalchemy** [comment]
Here's a fix.  In vendor/kartik-v/yii2-widget-select2/Select2.php after line 116 (parent::init()), add these two lines:
        $pluginId = ArrayHelper::getValue($this->pluginOptions, 'id', false);
        if ($pluginId) $this->options['id'] = $this->options['id']."-".$pluginId;

Then it seems to work.

**Comment by fredyns** [comment]
had you try to set [options->id] ?


			'options'		 => [
				'id' => 'element-id',
			],

**Comment by fnoopdogg** [comment]
Yes, I tried that.

**Comment by siliconalchemy** [comment]
Ops, yes I tried that.

**Comment by kartik-v** [comment]
Use yii renderAjax method to render content in your ajax rendered container when working with jQuery plugin based widgets (like Select2).

**Comment by xtinem** [comment]
I am having the same issue. 
Controller: 
return $this->renderAjax('repairs');

View:
 <div>
            <label class="control-label">Equipment</label>
        </div>
        <?= 
            Select2::widget([
                'name' => 'equipment_name',
                'data' => $equipments,
                'options' => ['placeholder' => 'Select an Equipment ...'],
                'pluginOptions' => [
                    'tags' => true,
                    'tokenSeparators' => [',', ' '],
                    'maximumInputLength' => 10
                ],
            ])
        ?>
        <div>
            <div>&nbsp;</div>
            <label class="control-label">Part</label>
        </div>
        <?= 
            Select2::widget([
                'name' => 'parts_name',
                'data' => $parts,
                'options' => ['placeholder' => 'Select a Part ...'],
                'pluginOptions' => [
                    'tags' => true,
                    'tokenSeparators' => [',', ' '],
                    'maximumInputLength' => 10
                ],
            ])
        ?>
![image](https://cloud.githubusercontent.com/assets/13326113/24464652/0fb3e6a8-14dd-11e7-9b09-50affcdcf686.png)

The 1st dropdown won't show but the 2nd drop down is okay. But this only happens when being called by modal / through Ajax. Did I miss anything?

**Comment by Mobydags** [comment]
You need to set unique id for each select2: 
```
'options' => [
           'id' => 'sel1',
           'multiple' => true,
 ]
'options' => [
           'id' => 'sel2',
           'multiple' => true,
 ]
```
